src/app/shared/actions/volume-actions/volume-resize/volume-resize.component.ts

```diff
@@ -43,19 +46,25 @@ export class VolumeResizeComponent implements OnInit, OnChanges {
     return maxRootCapability;
   }
 
+  public get volumeIsRoot(): boolean {
+    return isRoot(this.volume);
+  }
+
+  public get canResize(): boolean {
+    return (this.diskOfferings && this.diskOfferings.length > 0) || isRoot(this.volume);
+  }
+
   constructor(
     public dialogRef: MatDialogRef<VolumeResizeComponent>,
     public authService: AuthService,
   ) {}
 
-  public isCustomizedForVolume(diskOffering: DiskOffering): boolean {
-    if (diskOffering) {
-      return isCustomized(diskOffering);
-    }
-  }
-
   public ngOnInit(): void {
     this.newSize = this.volume.size / Math.pow(2, 30);
+    const availableStorage = Number(this.availableStorage);
+    this.availableStorageForResize = !isNaN(availableStorage)
```

Comment:
We can pass `number | null` in this component, so that we don't have to parse numbers manually. It will be just:

```
if (!availableStorage) {
  return null;
}

return availableStorage + this.newSize;

```